aws-sdk-core/lib/aws-sdk-core/cloudfront/url_signer.rb

```diff
@@ -0,0 +1,136 @@
+            end
+
+            # create a signed Amazon CloudFront URL
+            # @param [String] URL to sign
+            # @option param [String | Integer] UTC Unix timestamp
+            # @option param [String] JSON Policy
+            def signed_url(url, expires = nil, policy = nil)
+                url_sections = url.split('://')
+                if url_sections.length < 2
+                    raise ArgumentError, "Invaild URL:#{url}"
+                end
+
+                scheme = url_sections[0].gsub('*', '')
+                uri = "#{scheme}://#{url_sections[1]}"
+                signature = signature(resource(scheme, uri), expires, policy)
+                start_flag = (uri.include? '?') ? '&' : '?'
+
+                uri = "#{uri}#{start_flag}#{signature}"
+                if scheme == 'rtmp'
+                    rtmp_url(URI(uri))
+        		else
+        			uri
+                end
+            end
+
+            # create a relative signed URL for RTMP distribution
+            def rtmp_url(uri)
+                result = uri.path.gsub(' ', '/')
+        		result[0] = ''
+                if uri.query.nil?
+                    result
+        		else
+        			"#{result}?#{uri.query}"
+                end
+            end
+
+            # prepare resource for signing
+            def resource(scheme, url)
+                case scheme
+                when 'http', 'http*', 'https' then url
+                when 'rtmp'
+                    url_info = URI.parse(url)
+                    path = url_info.path
+        			path[0] = ''
+                    resource_content = "#{File.dirname(path)}"\
+                        "/#{File.basename(path)}".gsub(' ', '/')
+                    if url_info.query.nil?
+                    	resource_content
+        			else
+        				"#{resource_content}?#{uri.query}"
+                    end
+        		else
+        			raise ArgumentError, "Invaild URI scheme:#{scheme}."\
+                    	"Scheme must be one of: http, https, or rtmp."
+                end
+        	end
+
+            # create signed values that used to construct signed URLs
+            # @option param [String] CloudFront resource to which
+            #    this signature will grant access. Not used when a custom
+            #    policy is provided.
+            # @option param [String] UTC Unix timestamp used when
+            #    signing with a canned policy. Not required when passing
+            #    a custom policy.
+            # @option param [String] JSON policy. Use this option when
+            #    creating a signature for a custom policy.
+            def signature(resource = nil, expires = nil, policy = nil)
+                signature_content = []
+                if !policy.nil?
+                    policy = policy.gsub('/\s/s', '')
+                    signature_content << "Policy=#{encode(policy)}"
+                elsif !resource.nil? && !expires.nil?
```

Comment:
In Ruby, only `nil` and `false` evaluate to `false` in a boolean context. You can remove the not nil checks here.

```ruby
elsif resource && expires
```